projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/listeners/Function.java

```diff
@@ -982,6 +982,20 @@ private static int getAlign(long argument) {
         return (int) argument & (Long.SIZE - 1);
     }
 
+    private static Type getExtractedElementType(Type type) {
+        Type valType = type;
+        if (valType instanceof PointerType) {
```

Comment:
I have already wondered if `types.get(i++)` is wrong in all places where it appears. I just couldn't imagine that could have gone unnoticed. Will you fix that for the other instructions?